[Docs] SegmentedGroup: update component API docs to add SuperSelect and SuperSelectMultiple (HDS-4836) (#4018)
hashicorp/design-system: website/docs/components/segmented-group/partials/code/how-to-use.md

[[code-snippets/segmented-group-form]]

> ### Super Select

Reviewer: @KristinLBradley I guess the question is, should the new content focus on the fact that you can have a super select or that there are new layout options? I'm kind of leaning towards having the section be called layout and then we can explain the fullWidth/maxWidth args and use the super select as an example as to why you may use the new args.

Reviewer 2: I think that would be a good way to document it. @shleewhite 

You could add the new "Layout" section & include explanations of each of the new properties generally and then explain specifically how they are helpful when using a SuperSelect as you said.

> Use the yielded `SuperSelect` or `SuperSelectMultiple` segments when placing a Super Select inside a Segmented Group.

Reviewer: ```suggestion
Use the yielded `SuperSelectSingle` or `SuperSelectMultiple` segments when placing a Super Select inside a Segmented Group.
```

[[code-snippets/segmented-group-form]]

### Layout

Use `@isFullWidth` to make the Segmented Group fill the width of its parent container. Use `@maxWidth` to limit how wide the group can grow.

> We recommend using `@isFullWidth={{true}}` when placing a Super Select within the Segmented Group to prevent the Super Select from changing size as the user interacts with it.

Reviewer: ```suggestion
We recommend using `@isFullWidth` when placing a Super Select within the Segmented Group to prevent the Super Select from changing size as the user interacts with it.
```

Sorry, the brackets don't render on the site, so may as well simplify here 

<img width="281" height="47" alt="Screenshot 2026-08-12 at 7 30 53 AM" src="https://github.com/user-attachments/assets/08a67ed0-e1ad-4ead-81d1-59cf2e7f8cf0" />

[[code-snippets/segmented-group-form]]

### Layout

Use `@isFullWidth` to make the Segmented Group fill the width of its parent container. Use `@maxWidth` to limit how wide the group can grow.

We recommend using `@isFullWidth={{true}}` when placing a Super Select within the Segmented Group to prevent the Super Select from changing size as the user interacts with it.

> 

Reviewer 2: ```suggestion
```

Nit: It's normally best to use one blank line max to separate sections.

[[code-snippets/segmented-group-form]]

### Layout

Use `@isFullWidth` to make the Segmented Group fill the width of its parent container. Use `@maxWidth` to limit how wide the group can grow.

> These properties are especially useful when placing a Super Select inside a Segmented Group. In this case, use the yielded `SuperSelect` or `SuperSelectMultiple` segments.

Reviewer: ```suggestion
We recommend using `@isFullWidth={{true}}` when placing a Super Select within the Segmented Group to prevent the Super Select from changing size as the user interacts with it.
```

It is nice to explain why this is a necessary argument when youre using super selects

> If you want the Segmented Group to stretch to the width of its parent while keeping a maximum width, set `@isFullWidth={{true}}` and `@maxWidth` on the Segmented Group itself.

Reviewer: this line is kind of redundant now that line 19 is there.
